packer/grapl-service/image.pkr.hcl

```diff
@@ -0,0 +1,163 @@
+packer {
+  required_version = ">= 0.12.0"
+}
+
+variable "build_ami" {
+  description = "Whether or not to actually build an AMI. Set to `false` for doing testing"
+  type        = bool
+  default     = true
+}
+
+variable "ami_name_prefix" {
+  type    = string
+  default = "grapl-service"
+}
+
+variable "aws_region" {
+  type    = string
+  default = "us-east-1"
+}
+
+variable "consul_module_version" {
+  type    = string
+  default = "v0.10.1"
+}
+
+variable "consul_version" {
+  type    = string
+  default = "1.9.6"
+}
+
+variable "nomad_version" {
+  type    = string
+  default = "1.1.1"
+}
+
+variable "terraform_aws_nomad_version" {
+  type    = string
+  default = "v0.9.1"
+}
+
+variable "git_sha" {
+  description = "The git SHA of the commit the AMI is being generated from. If present, will be used to tag the AMI."
+  type        = string
+  default     = env("GIT_SHA")
+}
+
+variable "git_branch" {
+  description = "The git branch the AMI is being generated from. If present, will be used to tag the AMI."
+  type        = string
+  default     = env("GIT_BRANCH")
+}
+
+variable "buildkite_build_number" {
+  description = "The build number of the Buildkite pipeline this AMI was built in. If present, will be used to tag the AMI."
+  type        = string
+  default     = env("BUILDKITE_BUILD_NUMBER")
+}
+
+locals {
+  # These are various metadata tags we can add to the resulting
+  # AMI. If any are unset (like the Buildkite build number, if
+  # building outside of Buildkite), those tags will be filtered out
+  # and not added to the AMI.
+  tags_from_vars = {
+    "git:sha"                = "${var.git_sha}"
+    "git:branch"             = "${var.git_branch}"
+    "buildkite:build_number" = "${var.buildkite_build_number}"
+  }
+}
+
+locals {
+  formatted_timestamp = formatdate("YYYYMMDDhhmmss", timestamp())
+  copy_ami_to_regions = [
+    "us-east-2",
+    "us-west-1",
+    "us-west-2",
+  ]
+}
+
+data "amazon-ami" "amazon-linux-2-x86_64" {
+  filters = {
+    architecture                       = "x86_64"
+    "block-device-mapping.volume-type" = "gp2"
```

Comment:
Are the quotes necessary here? Or does this need to be a nested block?

Reply:
not necessary, but block doesn't work

Reply:
actually, turns out it is necessary _according to the linter_.